src/main/java/frc/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,129 @@
+
+public class LauncherSubsystem extends SubsystemBase {
+  private final CANSparkMax mIndexer = new CANSparkMax(LauncherConstants.INDEXER_ID, MotorType.kBrushless);
+
+  private final TalonFX mMainFlywheel = new TalonFX(LauncherConstants.MAIN_FLYWHEEL_ID);
+  private final TalonFX mRollerFlywheel = new TalonFX(LauncherConstants.ROLLER_FLYWHEEL_ID);
+
+  private final DigitalInput mBeamBreak = new DigitalInput(LauncherConstants.BEAMBREAK_ID);
+
+  private double mDesiredVelocityMain = 0;
+  private double mDesiredVelocityRoller = 0;
+
+  private LauncherMode mLauncherMode = LauncherMode.OFF;
+
+  private boolean mLauncherReady = false;
+
+  public LauncherSubsystem() {
+    resetEncoders();
+    configureDevices();
+  }
+  
+  public void configureDevices() {
+    Slot0Configs slotConfigs = new Slot0Configs();
+
+    slotConfigs.kP = LauncherConstants.LAUNCHER_PID_CONTROLLER.getP();
+    slotConfigs.kI = LauncherConstants.LAUNCHER_PID_CONTROLLER.getI();
+    slotConfigs.kD = LauncherConstants.LAUNCHER_PID_CONTROLLER.getD();
+    slotConfigs.kS = LauncherConstants.LAUNCHER_FF_CONTROLLER.ks;
+    slotConfigs.kV = LauncherConstants.LAUNCHER_FF_CONTROLLER.kv;
+
+    mMainFlywheel.getConfigurator().apply(slotConfigs);
+    mRollerFlywheel.getConfigurator().apply(slotConfigs);
+  }
+
+  @Override
+  public void periodic() {
+    double actualMainVelocity = mMainFlywheel.getVelocity().getValueAsDouble();
+    double actualRollerVelocity = mRollerFlywheel.getVelocity().getValueAsDouble();
+
+    mLauncherReady = Math.abs(Math.abs(mDesiredVelocityMain) - Math.abs(actualMainVelocity)) <= LauncherConstants.FLYWHEEL_TOLERANCE &&
+                     Math.abs(Math.abs(mDesiredVelocityRoller) - Math.abs(actualRollerVelocity)) <= LauncherConstants.FLYWHEEL_TOLERANCE && 
+                     mDesiredVelocityMain != 0 && 
+                     mDesiredVelocityRoller != 0;
+            
+    runLauncher();
+  }
+
+  public void runIndexer(double velocity) {
+    mIndexer.set(velocity);
+  }
+  
+  public void runLauncher() {
+    //predicted velocity values
+    if (mLauncherMode == LauncherMode.SPEAKER) {
+      mDesiredVelocityMain = LauncherState.SPEAKER_DEFAULT.mainVelocity;
+      mDesiredVelocityRoller = LauncherState.SPEAKER_DEFAULT.rollerVelocity;
+    }
+    else if (mLauncherMode == LauncherMode.AMP) {
+      mDesiredVelocityMain = LauncherState.AMP_DEFAULT.mainVelocity;
+      mDesiredVelocityRoller = LauncherState.AMP_DEFAULT.rollerVelocity;
+    }
+    else if (mLauncherMode == LauncherMode.TRAP) {
+      mDesiredVelocityMain = LauncherState.TRAP_DEFAULT.mainVelocity;
+      mDesiredVelocityRoller = LauncherState.TRAP_DEFAULT.rollerVelocity;
+    }
+    else if (mLauncherMode == LauncherMode.OFF) {
+      mDesiredVelocityMain = 0;
+      mDesiredVelocityRoller = 0;
+    }
+
+    mMainFlywheel.setControl(new VelocityVoltage(mDesiredVelocityMain).withSlot(0));
+    mRollerFlywheel.setControl(new VelocityVoltage(mDesiredVelocityMain).withSlot(0));
```

Comment:
If you make one the follower, you don't have to do this. Alternatively, make sure to set inversions.